client/src/components/User/ThemeSelector.vue

```diff
@@ -0,0 +1,104 @@
+<script setup>
+import { useCurrentTheme } from "@/composables/user";
+import { useConfig } from "@/composables/config";
+import { computed, watch, ref } from "vue";
+import { safePath } from "@/utils/redirect";
+const { currentTheme, setCurrentTheme } = useCurrentTheme();
+const { config, isLoaded } = useConfig();
+const show = ref(false);
+const currentValue = computed({
+    get: () => {
+        return currentTheme;
+    },
+    set: (theme) => {
+        setCurrentTheme(theme);
+    },
+});
+function getBackground(themeDetails) {
+    const background = themeDetails["--masthead-color"];
+    if (background) {
+        return { background };
+    }
+    return {};
+}
+function getLogo(themeDetails) {
+    return themeDetails["--masthead-logo-img"] ?? config.value.logo_src;
+}
+function getStyle(themeDetails, variantKey) {
+    const background = themeDetails[`--masthead-link-${variantKey}`];
+    const color = themeDetails[`--masthead-text-${variantKey}`];
+    const styles = {};
+    if (background) {
+        styles["background"] = background;
+    }
+    if (color) {
+        styles["color"] = color;
+    }
+    return styles;
+}
```

Comment:
Concerning the functions `getBackground` and `getStyle`: The css varaibles can be used directly inside the css classes. There is no need to map them manually using javascript.

Reply:
Thank you so much for your comments. Can you explain this a little bit more i.e. are you suggesting to use custom scss-functions instead of js? We have to keep in mind that the list of themes is dynamic and that theme attributes should properly default if not specified in the theme config.

Reply:
The point of css variables is to be runtime assignable. What you are doing here is essentially what css does out of the box. No custom scss functions needed, no js needed. In the css block below, use css variables instead of scss variables, and then set the variables via the style prop on the parent div `default-theme-masthead`.

Reply:
Awesome. Thank you.